test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,182 @@
+/*!
+ * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+const confirm = jest.fn().mockResolvedValue(true);
+jest.mock('cli-ux', () => {
+  // Mocking needs to happen before the command is imported
+  const cli = jest.requireActual('cli-ux');
+  return {
+    ...cli,
+    default: {
+      ...cli.default,
+      confirm
+    }
+  };
+});
+
+const warn = jest.fn(message => console.log('MOCKED WARNING: ', message));
+jest.mock('@oclif/command', () => {
+  const command = jest.requireActual('@oclif/command');
+  command.Command.prototype.warn = warn;
+  return command;
+});
+
+import { GeneratorOptions as GeneratorOptionsSDK, generatorOptionsCli as generatorOptionsSDK } from '@sap/cloud-sdk-generator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { FlagsParsed } from '../src/commands/generate-vdm';
+import * as foo from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(foo, 'toGeneratorSDK');
```

Comment:
Why not put this in the `beforeEach` and avoid calling `mockClear`

Reply:
Could also use: 
``` ts
beforeEach(() => {
  foo.toGeneratorSDK = jest.fn();
});
```
instead of spyOn

Reply:
For some positive generator tests I need the implementation. That is why I used spy on.